lib/cdo/script_constants.rb

```diff
@@ -275,27 +270,21 @@ def self.teacher_dashboard_name(script)
 
   # Sections can be assigned to both courses and scripts. We want to make sure
   # we give teacher dashboard the same information for both sets of assignables.
-  # We also expect to be in a mixed world for a time where this info is asked for
-  # both by dashboard and pegasus, and we want to keep that in sync. We accomplish
-  # most of that through this shared method, leaving it to dashboard/pegasus
-  # to take care of translating name/cateogry (with the unfortunate effect that
-  # we could have two different translations).
-  # @param course_or_script [Course|Script] A row object from either our courses
-  #   or scripts dashboard db tables.
-  # @param hidden [Boolean] True if the passed in item is hidden
+  # @param script [Course|Script] A row object from our course or script dashboard db table.
   # @return {AssignableInfo} without strings translated
-  def self.assignable_info(course_or_script)
-    name = ScriptConstants.teacher_dashboard_name(course_or_script[:name])
-    first_category = ScriptConstants.categories(course_or_script[:name])[0] ||
-        OTHER_CATEGORY_NAME
+  def self.assignable_info(course_or_script, full_course: false)
+    category = full_course ?
```

Comment:
@davidsbailey i lost your [previous comment](https://github.com/code-dot-org/code-dot-org/pull/28906#discussion_r290089459), but +1 for simplifying this! 

i've updated the logic so that `category` will be `nil` for a `full_course`. otherwise, if we left the way category was assigned previously, we would end up with:
```
category = ScriptConstants.categories(course_or_script[:name])[0] || OTHER_CATEGORY_NAME
# category == OTHER_CATEGORY_NAME for all full courses
```

let me know if you think there are any other things we can simplify/clarify in this area!

Reply:
This looks much better, thanks for improving!